zebra-chain/src/amount.rs

```diff
@@ -601,20 +601,19 @@ mod test {
         let another_one = Amount::<NonNegative>::try_from(1)?;
         let zero = Amount::<NonNegative>::try_from(0)?;
 
-        let hash_set: HashSet<Amount<NonNegative>, RandomState> =
-            HashSet::from_iter([one].iter().cloned());
```

Comment:
We can remove `iter::FromIterator` now that we're not using `from_iter` here anymore

Reply:
Oops, I must have missed that warning when I pushed the last change.